### PR TITLE
Support for OpenCV4.1.1

### PR DESCRIPTION
**Pull request by kumaakh**
resolves https://github.com/cpp-pm/hunter/issues/4
This pull request  creates a hunter package for openCV 4.1.1

### CODE REVIEW

File: CMakeLists.txt
Comment (bkotzz) [inline CMakeLists.txt]: We can also use the `cpp-pm` release here https://github.com/cpp-pm/hunter/releases/tag/v0.23.214, so that people have access to the latest `OpenCV-Extra` release that you are adding in https://github.com/cpp-pm/hunter/pull/8

File: 3rdparty/ippicv/ippicv.cmake
Comment (bkotzz) [inline 3rdparty/ippicv/ippicv.cmake:41]: Do we care to wrap the other `hunter_add_package` calls in `if(HUNTER_ENABLED)`? It's not strictly necessary in the fork as the fork is meant to be used for hunter anyways

File: CMakeLists.txt
Comment (kumaakh) [reply CMakeLists.txt]: Agree, I'll re run ci with this change to test if that breaks anything here.

File: 3rdparty/ippicv/ippicv.cmake
Comment (kumaakh) [reply 3rdparty/ippicv/ippicv.cmake:41]: I am a noob but I guess the main advantage of that would be if these patches were merged to upstream, they would work for both HUNTER_ENABLED and !HUNTER_ENABLED cases

Comment (rbsheth) [reply 3rdparty/ippicv/ippicv.cmake:41]: It would be nice to merge upstream, but that may or may not happen. Regardless, having the `HUNTER_ENABLED` blocks could help us patch in the future.

File: cmake/OpenCVFindLibsVideo.cmake
Comment (rbsheth) [inline cmake/OpenCVFindLibsVideo.cmake]: Does this break gstreamer support?

File: cmake/OpenCVFindLibsVideo.cmake
Comment (rbsheth) [inline cmake/OpenCVFindLibsVideo.cmake]: Were these `CHECK_MODULE` macros added by a previous Hunterization?

File: cmake/OpenCVFindLibsVideo.cmake
Comment (rbsheth) [inline cmake/OpenCVFindLibsVideo.cmake]: Nit: remove extra space

File: cmake/OpenCVFindLibsVideo.cmake
Comment (rbsheth) [inline cmake/OpenCVFindLibsVideo.cmake]: We don't need this file anymore? Does this limit the `videoio` module's functionality?

File: CMakeLists.txt
Comment (rbsheth) [inline CMakeLists.txt:109]: :+1:

File: cmake/OpenCVFindLibsVideo.cmake
Comment (kumaakh) [reply cmake/OpenCVFindLibsVideo.cmake]: Here is a pointer to the upstream commit where this script was more or less removed https://github.com/opencv/opencv/commit/73959fed4581556d8e6b770987c819adfacceb87#diff-147a2d6c0a28c306ecf5078d84c5eb84 

I fixed my intermediate commits to now be in sync with the upstream file:
https://github.com/opencv/opencv/blob/4.1.1/cmake/OpenCVFindLibsVideo.cmake

**Comment by kumaakh** [comment]
AFAIK I have addressed all comments by updating the change set. I have also re run CI jobs to make sure nothing has been lost in transit. It will be nice to have another review.

File: cmake/OpenCVFindLibsGrfmt.cmake
Comment (bkotzz) [inline cmake/OpenCVFindLibsGrfmt.cmake:232]: NIT: This isn't wrapped by HUNTER_ENABLED

File: cmake/OpenCVFindLibsGrfmt.cmake
Comment (bkotzz) [inline cmake/OpenCVFindLibsGrfmt.cmake:90]: NIT: This isn't wrapped by HUNTER_ENABLED

File: cmake/OpenCVFindLibsGrfmt.cmake
Comment (bkotzz) [inline cmake/OpenCVFindLibsGrfmt.cmake:34]: NIT: This should be inside the `IF(HUNTER_ENABLED)` below

File: cmake/OpenCVFindLibsGrfmt.cmake
Comment (bkotzz) [inline cmake/OpenCVFindLibsGrfmt.cmake:9]: NIT: This isn't wrapped by HUNTER_ENABLED

File: CMakeLists.txt
Comment (bkotzz) [inline CMakeLists.txt:106]: NIT: This isn't wrapped by HUNTER_ENABLED

**Review by bkotzz — APPROVED** [review]
Looks good. If you are interested in wrapping everything with HUNTER_ENABLED such that you can try and get the patch in upstream, I've pointed out the places that are currently missed, but it's not that important that they get addressed in this fork

**Review by rbsheth — APPROVED** [review]
LGTM, see @bkotzz's comments if you want to try to get this upstream.

**Comment by rbsheth** [comment]
Released: https://github.com/cpp-pm/opencv/releases/tag/v4.1.1-p0

**Pull request merged**
